src/api/createRequest.js

```diff
@@ -0,0 +1,148 @@
+import axios from 'axios'
+
+// default headers are for json api
+const defaultHeaders = {
+  'Content-Type': 'application/vnd.api+json',
+  Accept: 'application/vnd.api+json',
+}
+
+/*
+ * @param {String} rootURL
+ * @param {String} apiNamespace
+ * @param {String} resource
+ * @param {Object} headers - these will default to json api headers
+ * @returns {Object} instance of request
+ */
+const createRequest = ({ rootURL, apiNamespace, resource, headers = defaultHeaders }) => {
+  const baseURL = `${rootURL}/${apiNamespace}`
+  const api = axios.create({ baseURL, headers })
+
+  /*
+   * @param {Any} value
+   * @returns boolean
+   */
+  const isObject = (value) => {
+    return value && typeof value === 'object' && value.constructor === Object
+  }
+
+  /*
+   * @param {Any} value
+   * @returns boolean
+   */
+  const isString = (value) => {
+    return (typeof value === 'string' || value instanceof String) && value.length > 0
+  }
```

Comment:
I think both of these can be pulled out of the constructor, and just be defined in the file, otherwise you're creating a new function for each request.

Reply:
Yes. I did think about this at the start but wanted to see where we ended up first.